### ISSUE TITLE
Google Drive integration failing - Error 400: invalid_request

### ISSUE THREAD

**Issue by soapster** [Ylianst/MeshCentral#4358]
**Describe the bug**
Trying to connect to Google drive, for autobackup

**To Reproduce**
1. Go to My server --> General --> Google Drive Backup
2. Click on The link to the Google Drive Console
3. Get My client ID and secret. Enter into Mesh prompt.
4. Click link to Brows to this URL
5. See error

**Expected behavior**
Expect to see a login prompt to confirm connection to my account.

**Screenshots**
![18-42 -19](https://user-images.githubusercontent.com/8184435/182486295-57a01c71-5418-46a2-b9de-35ee58aace66.jpg)

![18-42 -41](https://user-images.githubusercontent.com/8184435/182486319-90da4597-34ff-4eac-bd05-8b99a2f65362.jpg)

![18-28 -00](https://user-images.githubusercontent.com/8184435/182486326-9f681f4f-834a-4393-a646-4409011b932c.jpg)

**Server Software (please complete the following information):**
 - OS: Ubuntu
 - Virtualization: No
 - Network: Home LAN - Gigabit fiber
 - Version: 1.0.63 (also happened before the upgrade)
 - Node: 12.22.9

**Your config.json file**
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "_MongoDb": "mongodb://localhost:27017/meshcentral",
    "cert": "sub.domain.com",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "webrtc": true,
    "AgentSignLock": true,
    "allowHighQualityDesktop": true,
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_DesktopMultiplex": true,
    "_crowdsec": {
      "url": "http://localhost:8080",
      "apiKey": "theapikey - cant make this work yet",
      "fallbackremediation": "captcha"
    },
    "_redirAliasPort": 80,
    "autoBackup": {
      "backupIntervalHours": 24,
      "keepLastDaysBackup": 14,
      "zipPassword": "zippassword",
      "googleDrive": {
        "folderName": "Mesh-Backups",
        "maxFiles": 24
      }
    }
  },
  "domains": {
    "": {
      "title": "Soapy's Mesh",
      "_title2": "Remote Support",
      "mstsc": true,
      "ssh": true,
      "_minify": true,
      "newAccounts": false,
      "_userNameIsEmail": true
    }
  },
  "letsencrypt": {
    "email": "emailaccount",
    "names": "sub.domain.com",
    "skipChallengeVerification": false,
    "production": true
 }
}

**Comment by soapster** [comment]
This used to work fine. I was just randomly checking my backups, and found it hasn't updated since June 4th (I don't check often).

The only issue I had before was that the "app" on the google console was in testing. Once I "published" it, it just kept working.

I'm going to check the one at my work, to see if it is still working.

**Comment by soapster** [comment]
Very strange... The server at work is backing up normally, without issue... I'll start poking around in the google console, and such. Will post any findings.

**Comment by soapster** [comment]
Ok, so it seems this is an issue with google????  I updated my node to v16 and it still failed.
So I dug into the Google console some more.  It looks like my open project was started by another app... In my case, pokemon go??? Maybe?   So what I did was start a new project, specifically for Mesh.  Added the OAuth consent screen, and created new credentials.  Ran through the process to connect Mesh to it, and it went without a hitch.

Instead of Deleting this issue, I figured I would post details in case someone else has a similar issue.

Closing.  (I wish I did these steps before even submitting the issue)

**Comment by omriasta** [comment]
Hmmm...I just went through the same thing. My setup worked for 10 days since I set it up and stopped. Looking around for the error I got (invalid grant I believe) pointed to the fact that my app was set to testing. Once I set it to Production, I got the same error you got. I recreated credentials while the app was set to production and got the same error.
The last thing I did was keep it in testing, authorize meshcentral and then switch it to production. Currently it is working but I guess I'll have to wait for 10 days to check if the token will expire again.....

**Comment by soapster** [comment]
So it looks like my backups stopped again. Invalid Grant.  I don't understand. This is still working at my workplace, without issue.
I'll have to go over my mesh and drive setup at work, to see what I am missing. Will report back what I find... If I find any differences.  If anyone with experience with this integration could chime in, it would be greatly appreciated.

**Comment by soapster** [comment]
Ok, I must have been drinking when I fix it last time. My "app" was still in testing.  I did compare my work setup to my home, and I did notice a difference.
Under the Credentials section of the Google API deal...
![18-25 -02](https://user-images.githubusercontent.com/8184435/185505826-7e16ccea-fc69-4500-bcc6-ff8759c68a58.jpg)
I was missing a service account.  I added that. 
Then I re-authorized  mesh backup. 
Finally, I moved my app out of testing. 

Will see what happens in a week or 2.

**Comment by omriasta** [comment]
Hmmm...I just tried that and it worked for about a week or so and I am getting this again:
`GoogleDrive error: Error: invalid_grant`

**Comment by omriasta** [comment]
@soapster is this still working for you? I tried all of the above and it still stops working after a few days.....

**Comment by soapster** [comment]
@omriasta Nope. Well, I did get it to authenticate, but it stopped working again.  The one I have at work is still working fine though.  I can't tell why that one is working.
Re-authorizing it does work, if it is in testing, but not if the app is published. Then you can switch it to published after it is authenticated.

I'm out of ideas.
